cast/src/lib.rs

```diff
@@ -149,11 +152,38 @@ pub async fn wait_for_tx(provider: &JsonRpcClient<HttpTransport>, tx_hash: Field
                 sleep(Duration::from_secs(5));
                 true
             }
-            TransactionStatus::AcceptedOnL2 | TransactionStatus::AcceptedOnL1 => false,
+            TransactionStatus::AcceptedOnL2 | TransactionStatus::AcceptedOnL1 => {
+                return Ok("Transaction accepted")
+            }
             TransactionStatus::Rejected => {
-                println!("{}", style("Transaction has been rejected").red());
-                false
+                return Err(anyhow!(
+                    "Transaction has been rejected. Check if max-fee argument is high enough"
```

Comment:
Is this most common reason? Maybe we can get a real reason somehow? If not, maybe we should remove this, so it is not misleading?

Reply:
Unfortunately, we can't :/ I would really like to get it from the net but it is not possible in the RPC. I can remove it